tests/unit/test_mode_functions.py

```diff
@@ -0,0 +1,55 @@
+import unittest
+from unittest.mock import patch
+
+from src.component_managers.astt_comp_manager import (
+    ASTTComponentManager,
+)
+from src.component_managers.dish_modes import Mode
+
+
+class TestAntennaModes(unittest.TestCase):
+    @patch("src.component_managers.astt_comp_manager.canopen.Network")
+    @patch(
+        "src.component_managers.astt_comp_manager.canopen.RemoteNode"
+    )
+    def setUp(self, mock_RemoteNode, antenna_node):
+        # Create an instance of ASTTComponentManager
+        self.manager = ASTTComponentManager()
+
+        # Mock the PLC node
+        mock_plc_node = mock_RemoteNode.return_value
+        self.manager.antenna_node = mock_plc_node
+
+    @patch.object(
+        ASTTComponentManager,
+        "get_antenna_mode",
+        return_value=Mode.POINT,
+    )  # noqa
+    def test_set_point_mode(self, mock_get_antenna_mode):
+        self.manager.set_point_mode()
+        mode = self.manager.get_antenna_mode()
+        self.assertEqual(mode, Mode.POINT)
+
+    @patch.object(
+        ASTTComponentManager,
+        "get_antenna_mode",
+        return_value=Mode.IDLE,
+    )  # noqa
+    def test_set_idle_mode(self, mock_get_antenna_mode):
+        self.manager.set_idle_mode()
+        mode = self.manager.get_antenna_mode()
+        self.assertEqual(mode, Mode.IDLE)
```

Comment:
This test is testing against a mock object. Therefore it does not prove functionality.
 You have mocked get_antenna_mode, therefore the assert statement will pass automatically because the mocked value and the test value are the same. Please find the article on mocking practices: https://www.telerik.com/blogs/mocking-best-practices

Reply:
The functions being tested have no return so they'll always return a None when being tested hence this implementation. Either we have a return in the functions or we delete the test.